Dockerfile

```diff
@@ -0,0 +1,50 @@
+# Build stage
+FROM node:22-slim AS builder
+
+# Set working directory
+WORKDIR /app
+
+# Copy package files
+COPY package*.json ./
+
+# Install dependencies
+RUN npm ci
+
+# Copy source code
+COPY . .
+
+# Build the application
+RUN npm run build
+
+# Runtime stage
+FROM node:22-slim AS runtime
+
+# Install tini
+RUN apt-get update && \
+    rm -rf /var/lib/apt/lists/*
+
+# Create non-root user
+RUN groupadd -r recoil && useradd -r -g recoil recoil
+
+# Set working directory
+WORKDIR /app
+
+# Copy built application from builder
+COPY --from=builder /app/dist ./dist
+COPY --from=builder /app/package*.json ./
+
+# Install production node dependencies
+RUN npm ci --omit dev --omit optional
```

Comment:
Let's not skip optional, we should have `bufferutil` installed. It's optional only in in case somebody tries to do run it on some platform/node version where `bufferutil` doesn't have pre-build binary and fails to compile.